amy/extrequests/forms.py

```diff
@@ -440,6 +453,31 @@ def institution_label_from_instance(obj):
         essentially works just like `Model.__str__`."""
         return "{}".format(obj.fullname)
 
+    @feature_flag_enabled("ENFORCE_MEMBER_CODES")
+    def clean_membership_info(self, request: HttpRequest) -> dict:
```

Comment:
If form method is called `clean_xyz`, then I assume it is run for validation of a single field `xyz`, which is not a case here. I'd rename to `clean_member_code` if we have access to `self.cleaned_data` in per-field validation, otherwise rename to `validate_member_code`.

If we do have access to `self.cleaned_data` in per-field validation then calling this method in `clean()` will not be necessary.

Reply:
Per-field validators do have access to `self.cleaned_data`, but the `request` argument is needed here for `@feature_flag_enabled` to work, so I think this has to be called manually from `clean()`.

This raises a side question: do we plan to remove feature flags once they're enabled and stable in production? If so I guess I should plan for some refactoring/cleanup at the end of this project.

Reply:
We could remove them in the long-term, but for now we can use the database to set the feature flag dynamically to always be enabled (we discussed this a few weeks back with one of my PRs).